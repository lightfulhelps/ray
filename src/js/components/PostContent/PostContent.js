// @flow
import * as React from 'react';
import classNames from 'classnames';
import Dotdotdot from 'react-dotdotdot';

type Props = {
  children: string,
  className?: string,
  lines?: number,
};

const PostContent = ({ children, className, lines = 5 }: Props) => {
  const classes = classNames(className, 'post-content', 'my-1 mx-2 font-weight-light');

  return (
    <div className={classes}>
      <Dotdotdot clamp={lines}>{children}</Dotdotdot>
    </div>
  );
};

export default PostContent;
