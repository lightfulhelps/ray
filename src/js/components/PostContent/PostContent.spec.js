import React from 'react';
import { shallow } from 'enzyme';
import PostContent from './PostContent';

const setup = (overrides = {}) => {
  const props = {
    children:
      'Buttle UK helped more than 3,000 vulnerable families buy beds for their children last year. It fears thousands more across the UK may lack a bed of their own, leading to problems concentrating in school. The government said its welfare reforms were "supporting those who need it most".',
    ...overrides,
  };
  const wrapper = shallow(<PostContent {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('<PostContent />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the className prop', () => {
    const { wrapper } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('post-content')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle the children prop', () => {
    const { wrapper, props } = setup();

    expect(wrapper.childAt(0).prop('children')).toEqual(props.children);
  });

  it('should handle the lines prop', () => {
    const { wrapper } = setup({ lines: 3 });

    expect(wrapper.childAt(0).prop('clamp')).toEqual(3);
  });
});
