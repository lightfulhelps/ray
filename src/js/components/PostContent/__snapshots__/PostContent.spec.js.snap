// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostContent /> should render 1`] = `
<div
  className="post-content my-1 mx-2 font-weight-light"
>
  <Dotdotdot
    clamp={5}
    tagName="div"
    truncationChar="â€¦"
    useNativeClamp={true}
  >
    Buttle UK helped more than 3,000 vulnerable families buy beds for their children last year. It fears thousands more across the UK may lack a bed of their own, leading to problems concentrating in school. The government said its welfare reforms were "supporting those who need it most".
  </Dotdotdot>
</div>
`;
